src/Analysis/GWASResults/SharedComponents/Diagrams/lib/pheweb_plots.js

```diff
@@ -0,0 +1,608 @@
+/**
+ * PheWeb Manhattan and QQ plots
+ * Duplicated from https://github.com/statgen/locuszoom-hosted/blob/c153d3ba465be5b171d073c0aa785e86c1cde109/assets/js/util/pheweb_plots.js
+ * MIT LICENSE: https://github.com/statgen/locuszoom-hosted/blob/d726fb23477e36d244c43be87d84b5ed3e1725e7/LICENSE, https://github.com/statgen/pheweb/blob/76f0d0e32ae72e51bc4b259ce4b16edfd653601a/LICENSE
+ */
+
+import * as d3 from 'd3-selection';
+import * as d3Array from 'd3-array';
+import * as d3Scale from 'd3-scale';
+import * as d3Axis from 'd3-axis';
+import * as d3Format from 'd3-format';
+import d3Tip from 'd3-tip';
+import {memoize, property, range, some, sortBy, template} from 'lodash';
+
+/* eslint camelcase: 0 */ // --> OFF
+/* eslint no-unused-vars: 0 */ // --> OFF
+/* eslint max-len: 0 */ // --> OFF
+/* eslint no-plusplus: 0 */ // --> OFF
+/* eslint indent: 0 */ // --> OFF
+/* eslint spaced-comment: 0 */ // --> OFF
+/* eslint block-spacing: 0 */ // --> OFF
+/* eslint prefer-arrow-callback: 0 */ // --> OFF
+/* eslint space-before-function-paren: 0 */ // --> OFF
+/* eslint no-multiple-empty-lines: 0 */ // --> OFF
+/* eslint comma-spacing: 0 */ // --> OFF
+/* eslint brace-style: 0 */ // --> OFF
+/* eslint vars-on-top: 0 */ // --> OFF
+/* eslint quote-props: 0 */ // --> OFF
+/* eslint no-multi-spaces: 0 */ // --> OFF
+/* eslint padded-blocks: 0 */ // --> OFF
+/* eslint object-curly-spacing: 0 */ // --> OFF
+/* eslint object-curly-newline: 0 */ // --> OFF
+/* eslint eqeqeq: 0 */ // --> OFF
+/* eslint space-in-parens: 0 */ // --> OFF
+/* eslint no-floating-decimal: 0 */ // --> OFF
+/* eslint prefer-template: 0 */ // --> OFF
+/* eslint newline-per-chained-call: 0 */ // --> OFF
+/* eslint object-shorthand: 0 */ // --> OFF
+/* eslint prefer-rest-params: 0 */ // --> OFF
+/* eslint prefer-destructuring: 0 */ // --> OFF
+/* eslint wrap-iife: 0 */ // --> OFF
+/* eslint arrow-parens: 0 */ // --> OFF
+
+// Based on https://github.com/statgen/locuszoom-hosted/commit/4c72340981c3e07733fedd14cb3ff479468db5d1#diff-663dda1af63699b00348953ced366f82022b3675c7df72276425a75f27312994R148, and adjusted a bit:
+const tooltip_underscoretemplate = '<b><%- d.chrom %>:<%- d.pos.toLocaleString() %> <%- (d.ref && d.alt) ? (d.ref + "/" + d.alt) : "" %></b><br>p-value: <%- d.pval %><br>Nearest gene(s): <%- d.nearest_genes %>';
+const urlprefix = 'mylocuszoom-url-prefix';
+
+// NOTE: `qval` means `-log10(pvalue)`.
+function fmt(format) {
+    // convenience functions- from pheweb common.js
+    const args = Array.prototype.slice.call(arguments, 1);
+    return format.replace(/{(\d+)}/g, function(match, number) {
+        return (typeof args[number] != 'undefined') ? args[number] : match;
+    });
+}
+
+function create_gwas_plot(variant_bins, unbinned_variants_in) {
+    // FIXME: Replace global variables with options object
```

Comment:
If this is your comment, I'd suggest creating a ticket for it and removing the FIXME comment.

Reply:
not my comment....see https://github.com/statgen/locuszoom-hosted/blob/c153d3ba465be5b171d073c0aa785e86c1cde109/assets/js/util/pheweb_plots.js#L23